src/app/AttributeAccessInterface.h

```diff
@@ -414,6 +414,25 @@ class AttributeAccessInterface
      */
     virtual CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) { return CHIP_NO_ERROR; }
 
+    /**
+     * Indicates the start of a series of list operations, this function will be called before first Write to the list.
```

Comment:
My point was that we need to clearly document the behavior: it's coalescing things that the client might be thinking of as separate writes into a single "transaction".

You know what the behavior is: you wrote the implementation.  I know what the behavior is: I read the implementation.  API consumers should not need to read the implementation.

Reply:
Added the descriptions about when will the functions be invoked.